File: Week3/Code/Vectorize1.R
#!/usr/bin/env R 

#Example of vectorization

M <- matrix(runif(1000000),1000,1000)

SumAllElements <- function(M){
  Dimensions <- dim(M)
  Tot <- 0
  for (i in 1:Dimensions[1]){
    for (j in 1:Dimensions[2]){
      Tot <- Tot + M[i,j]
    }
  }
  return (Tot)
}
 
time1 = as.numeric(system.time(res2<-SumAllElements(M))[3])
time1_f = format(round(time1, 4), nsmall = 4)
time2 = as.numeric(system.time(res2<-sum(M))[3])
time2_f = format(round(time2, 4), nsmall = 4)
cat(' -------------  -----------|------------', '\n')
cat(" Vectorize1.R  |", time1, '\t\t', time2, ' |\n')

File: Week3/Code/Vectorize1.py
#!/usr/bin/env python3

'''Python version of Vectorize1.R'''

import numpy as np
import time

M = np.random.uniform(0,1,(1000,1000))

def SumAllElements(M):
    Dimensions = M.shape
    Tot = 0
    for i in np.arange(Dimensions[0]):
        for j in np.arange(Dimensions[1]):
            Tot = Tot + M[i,j]

    return Tot

print("                  Non-vect   Vectorized")
start = time.time()
SumAllElements(M)
end = time.time()
tot1 = end - start
start = time.time()
np.sum(M)
end = time.time()
tot2 = end - start
print('                -----------|------------')
print(" Vectorize1.py | {:.{}f}\t\t".format(tot1,4), "{:.{}f}".format(tot2,4),
        '|')


